grayscale.py

```diff
@@ -392,97 +398,88 @@ def brightness_sync(self, c):
         while self._state[Grayscale_StateIndex_ContrastChng] != 0xff:
             idle()
 
-    @micropython.native
+    @micropython.viper
     def _copy_buffers(self):
-        b1 = self.buffer1 ; b2 = self.buffer2
-        _b1 = self._buffer1 ; _b2 = self._buffer2 ; _b3 = self._buffer3
+        b1:ptr32 = ptr32(self.buffer1) ; b2:ptr32 = ptr32(self.buffer2)
+        _b1:ptr32 = ptr32(self._buffer1) ; _b2:ptr32 = ptr32(self._buffer2) ; _b3:ptr32 = ptr32(self._buffer3)
         i:int = 0
-        bs:int = self.buffer_size
-        while i < bs:
-            v1 = b1[i]
-            v2 = b2[i]
+        while i < 90:
+            v1:int = b1[i]
+            v2:int = b2[i]
             _b1[i] = v1 | v2
             _b2[i] = v2
             _b3[i] = v1 & v2
             i += 1
         self._state[Grayscale_StateIndex_CopyBuffs] = 0
 
 
-    @micropython.native
-    def _display_thread_frame(self, fn:int, post_frame_adj:ptr8, fb:ptr8):
-        t0:int = utime.ticks_us()
+    @micropython.viper
+    def _display_thread(self):
```

Comment:
This function is some seriously impressive optimized Viper coding man. I am in awe 😄

For more casual viewers it could maybe do with a few comments, like some broad strokes what's happening the different parts. I could try to add that, if you don't feel like it?

Reply:
Yeah I can add in some comments... I'll push that once I'm done, so please add more if you think that's needed.